### PR TITLE
Migrate transforms-full to TypeScript

### PR DESCRIPTION
**Pull request by captainsafia**
Closes https://github.com/nteract/nteract/issues/3589.

This should be merged after the other transforms are migrated.

### CODE REVIEW

File: packages/transforms-full/package.json [hunk 1]
```diff
@@ -33,14 +33,13 @@
...
-    "@babel/runtime-corejs2": "^7.0.0",
     "@nteract/jupyter-widgets": "^1.0.0-alpha.0",
     "@nteract/transform-dataresource": "^5.0.0-alpha.0",
     "@nteract/transform-geojson": "^4.0.0-alpha.0",
     "@nteract/transform-model-debug": "^4.0.0-alpha.0",
     "@nteract/transform-plotly": "^4.0.0-alpha.0",
     "@nteract/transform-vega": "^4.0.0-alpha.0",
     "@nteract/transforms": "^5.0.0-alpha.0",
-    "babel-runtime": "^6.26.0"
+    "vega": "^4.3.0"
```
Comment (rgbkrk) [inline packages/transforms-full/package.json]: Wait is this right? Aren't we depending on bundled Vega in the Vega transform?

**Review by rgbkrk — APPROVED** [review]

**Comment by rgbkrk** [comment]
Merge conflicts in here now

File: packages/transforms-full/package.json (hunk 1 above)
Comment (captainsafia) [reply packages/transforms-full/package.json]: I added it to include a type definition. I can remove it. It's a little overkill to include for one type.

**Comment by captainsafia** [comment]
OK. I fixed this up now.

**Comment by rgbkrk** [comment]
🎉

**Pull request merged**